### PR TITLE
OptionFactory.createHighlighted - Supporting spaces in search term

### PR DESCRIPTION
**Pull request by giladartzi**
This change allows us to highlight values, even if the search term contains spaces.

Having an option with the value of "New York" -
Before this change, if the user has typed "ne", the result would be "**Ne**w York" (which is perfectly fine), but if the user typed "Ne Y", the string will not be highlighted at all.
After this fix, for the "Ne Y" input, the result will be "**Ne**w **Y**ork"

### CODE REVIEW

File: packages/wix-ui-core/src/components/address-input/AddressInput.spec.tsx [hunk 1]
```diff
@@ -799,6 +799,36 @@ describe('AddressInput', () => {
...
+            driver.keyDown('e');
+            await waitForCond(() => driver.isContentElementExists());
+            const highlightedStrings = driver.optionAt(0).getHighlightedStrings();
+            expect(highlightedStrings).toEqual({
+                highlighted: ['Ne'],
+                nonHighlighted: ['1 East Broadway, ', 'w York, NY, USA']
+            });
+        });
+
+        it('should highlight text in dropdown, even if it is not continuous', async () => {
```
Comment (shlomitc) [inline packages/wix-ui-core/src/components/address-input/AddressInput.spec.tsx]: this functionality should be tested in `DropdownOption.spec.tsx`

File: packages/wix-ui-core/src/components/dropdown-option/DropdownOption.driver.ts [hunk 2]
```diff
@@ -13,5 +13,14 @@ export const dropdownOptionDriverFactory = ({element, eventTrigger}) => {
     isDisabled: () => element && domUtils.hasStyleState(element, 'disabled'),
     getText: () => element && element.textContent,
     getElement: () => element,
+    getHighlightedStrings: () => {
```
Comment (shlomitc) [inline packages/wix-ui-core/src/components/dropdown-option/DropdownOption.driver.ts]: please remove the `nonHighlighted`, return only the highlighted part

File: packages/wix-ui-core/src/components/address-input/AddressInput.spec.tsx (hunk 1 above)
Comment (giladartzi) [reply packages/wix-ui-core/src/components/address-input/AddressInput.spec.tsx]: Agreed. Moved to `DropdownOption.spec.tsx`

File: packages/wix-ui-core/src/components/dropdown-option/DropdownOption.driver.ts (hunk 2 above)
Comment (giladartzi) [reply packages/wix-ui-core/src/components/dropdown-option/DropdownOption.driver.ts]: Done

**Pull request merged**